src/multisig/multisigController.ts

```diff
@@ -0,0 +1,834 @@
+
+        return transaction;
+    }
+
+    /**
+     * Awaits the completion of a sign batch action
+     */
+    async awaitCompletedSignBatch(txHash: string): Promise<void> {
+        await this.transactionAwaiter.awaitCompleted(txHash);
+    }
+
+    /**
+     * Creates a transaction for signing and performing an action in one step
+     */
+    async createTransactionForSignAndPerform(
+        sender: IAccount,
+        nonce: bigint,
+        options: resources.ActionInput & BaseControllerInput,
+    ): Promise<Transaction> {
+        const transaction = this.multisigFactory.createTransactionForSignAndPerform(sender.address, options);
+
+        transaction.guardian = options.guardian ?? Address.empty();
+        transaction.relayer = options.relayer ?? Address.empty();
+        transaction.nonce = nonce;
+        this.setTransactionGasOptions(transaction, options);
+        transaction.signature = await sender.signTransaction(transaction);
+
+        return transaction;
+    }
+
+    /**
+     * Awaits the completion of a sign and perform action
+     */
+    async awaitCompletedSignAndPerform(txHash: string): Promise<void> {
+        await this.transactionAwaiter.awaitCompleted(txHash);
+    }
+
+    /**
+     * Creates a transaction for unsigning for outdated board members
+     */
+    async createTransactionForUnsignForOutdatedBoardMembers(
+        sender: IAccount,
+        nonce: bigint,
+        options: resources.UnsignForOutdatedBoardMembersInput & BaseControllerInput,
+    ): Promise<Transaction> {
+        const transaction = this.multisigFactory.createTransactionForUnsignForOutdatedBoardMembers(
+            sender.address,
+            options,
+        );
+
+        transaction.guardian = options.guardian ?? Address.empty();
+        transaction.relayer = options.relayer ?? Address.empty();
+        transaction.nonce = nonce;
+        this.setTransactionGasOptions(transaction, options);
+        transaction.signature = await sender.signTransaction(transaction);
+
+        return transaction;
+    }
+
+    /**
+     * Awaits the completion of an unsign for outdated board members action
+     */
+    async awaitCompletedUnsignForOutdatedBoardMembers(txHash: string): Promise<void> {
+        await this.transactionAwaiter.awaitCompleted(txHash);
+    }
+
+    /**
+     * Creates a transaction for performing a batch of actions
+     */
+    async createTransactionForPerformBatch(
+        sender: IAccount,
+        nonce: bigint,
+        options: resources.GroupInput & BaseControllerInput,
+    ): Promise<Transaction> {
+        const transaction = this.multisigFactory.createTransactionForPerformBatch(sender.address, options);
+
+        transaction.guardian = options.guardian ?? Address.empty();
+        transaction.relayer = options.relayer ?? Address.empty();
+        transaction.nonce = nonce;
+        this.setTransactionGasOptions(transaction, options);
```

Comment:
In the Python controllers, we do some additional logic: handling "sign with hash" and "setting options if guarded transaction". We should do this here, as well - but in another PR (separate from multisig feature, for all controllers), of course.